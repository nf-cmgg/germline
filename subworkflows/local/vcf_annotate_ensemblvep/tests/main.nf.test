nextflow_workflow {

    name "Test Subworkflow VCF_ANNOTATE_ENSEMBLVEP"
    script "../main.nf"
    workflow "VCF_ANNOTATE_ENSEMBLVEP"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/vcf_annotate_ensemblvep"
    tag "ensemblvep/download"
    tag "ensemblvep/vep"
    tag "tabix/tabix"
    tag "tabix/bgzip"
    tag "bcftools/pluginscatter"
    tag "bcftools/concat"
    tag "bcftools/sort"

    config "./nextflow.config"

    test("sarscov2 - ensemblvep") {

        setup {
            run("ENSEMBLVEP_DOWNLOAD") {
                script "../../../../modules/nf-core/ensemblvep/download"
                process {
                    """
                    input[0] = [
                        [id:"reference"],
                        "WBcel235",
                        "caenorhabditis_elegans",
                        "110"
                    ]
                    """
                }
            }
        }

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'custom_test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test2.vcf.gz', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test2.vcf.gz.tbi', checkIfExists: true),
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test3.vcf.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test3.vcf.gz.tbi', checkIfExists: true)
                    ]
                ])
                input[1] = [[],[]]
                input[2] = "WBcel235"
                input[3] = "caenorhabditis_elegans"
                input[4] = "110"
                input[5] = ENSEMBLVEP_DOWNLOAD.out.cache.map { meta, cache -> cache }
                input[6] = []
                input[7] = 5
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.out.vcf_tbi.collect { [it[0], "${file(it[1]).name},variantsMD5:${path(it[1]).vcf.variantsMD5}", file(it[2]).name] },
                    workflow.out.vep_reports.collect { it instanceof String ? file(it).name : it },
                    workflow.out.versions.collect { it instanceof String ? file(it).name : it }
                ).match()}
            )
        }
    }

    test("sarscov2 - ensemblvep - large chunks") {

        setup {
            run("ENSEMBLVEP_DOWNLOAD") {
                script "../../../../modules/nf-core/ensemblvep/download"
                process {
                    """
                    input[0] = [
                        [id:"reference"],
                        "WBcel235",
                        "caenorhabditis_elegans",
                        "110"
                    ]
                    """
                }
            }
        }

        when {
            workflow {
                """
                input[0] = Channel.of(                [
                    [ id:'custom_test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test2.vcf.gz', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test2.vcf.gz.tbi', checkIfExists: true),
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test3.vcf.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test3.vcf.gz.tbi', checkIfExists: true)
                    ]
                ])
                input[1] = [[],[]]
                input[2] = "WBcel235"
                input[3] = "caenorhabditis_elegans"
                input[4] = "110"
                input[5] = ENSEMBLVEP_DOWNLOAD.out.cache.map { meta, cache -> cache }
                input[6] = []
                input[7] = 100
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.out.vcf_tbi.collect { [it[0], "${file(it[1]).name},variantsMD5:${path(it[1]).vcf.variantsMD5}", file(it[2]).name] },
                    workflow.out.vep_reports.collect { it instanceof String ? file(it).name : it },
                    workflow.out.versions.collect { it instanceof String ? file(it).name : it }
                ).match()}
            )
        }
    }

    test("sarscov2 - ensemblvep - no scatter") {

        setup {
            run("ENSEMBLVEP_DOWNLOAD") {
                script "../../../../modules/nf-core/ensemblvep/download"
                process {
                    """
                    input[0] = [
                        [id:"reference"],
                        "WBcel235",
                        "caenorhabditis_elegans",
                        "110"
                    ]
                    """
                }
            }
        }

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'custom_test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test2.vcf.gz', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test2.vcf.gz.tbi', checkIfExists: true),
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test3.vcf.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/vcf/test3.vcf.gz.tbi', checkIfExists: true)
                    ]
                ])
                input[1] = [[],[]]
                input[2] = "WBcel235"
                input[3] = "caenorhabditis_elegans"
                input[4] = "110"
                input[5] = ENSEMBLVEP_DOWNLOAD.out.cache.map { meta, cache -> cache }
                input[6] = []
                input[7] = []
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.out.vcf_tbi.collect { [it[0], "${file(it[1]).name},variantsMD5:${path(it[1]).vcf.variantsMD5}", file(it[2]).name] },
                    workflow.out.vep_reports.collect { it instanceof String ? file(it).name : it },
                    workflow.out.versions.collect { it instanceof String ? file(it).name : it }
                ).match()}
            )
        }
    }
}
