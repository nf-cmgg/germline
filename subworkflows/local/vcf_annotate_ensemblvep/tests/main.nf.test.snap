{
    "sarscov2 - ensemblvep - large chunks": {
        "content": [
            [
                [
                    {
                        "groupSize": 1,
                        "groupTarget": {
                            "id": "custom_test",
                            "single_end": false
                        }
                    },
                    "custom_test.vcf.gz,variantsMD5:44ed24c4dc4223670a78ffea3c7459e",
                    "custom_test.vcf.gz.tbi"
                ]
            ],
            [
                "custom_test0_annotated.vep.vcf.gz_summary.html"
            ],
            [
                "versions.yml",
                "versions.yml",
                "versions.yml",
                "versions.yml",
                "versions.yml"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-20T14:31:50.569767686"
    },
    "sarscov2 - ensemblvep - no scatter": {
        "content": [
            [
                [
                    {
                        "id": "custom_test",
                        "single_end": false
                    },
                    "custom_test.vep.vcf.gz,variantsMD5:44ed24c4dc4223670a78ffea3c7459e",
                    "custom_test.vep.vcf.gz.tbi"
                ]
            ],
            [
                "custom_test.vep.vcf.gz_summary.html"
            ],
            [
                "versions.yml",
                "versions.yml"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-20T14:32:06.437006334"
    },
    "sarscov2 - ensemblvep": {
        "content": [
            [
                [
                    {
                        "groupSize": 2,
                        "groupTarget": {
                            "id": "custom_test",
                            "single_end": false
                        }
                    },
                    "custom_test.vcf.gz,variantsMD5:44ed24c4dc4223670a78ffea3c7459e",
                    "custom_test.vcf.gz.tbi"
                ]
            ],
            [
                "custom_test0_annotated.vep.vcf.gz_summary.html",
                "custom_test1_annotated.vep.vcf.gz_summary.html"
            ],
            [
                "versions.yml",
                "versions.yml",
                "versions.yml",
                "versions.yml",
                "versions.yml"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-20T14:31:32.997494596"
    }
}